app/src/main/java/com/babylon/wallet/android/presentation/account/settings/thirdpartydeposits/AccountThirdPartyDepositsViewModel.kt

```diff
@@ -105,29 +110,29 @@ class AccountThirdPartyDepositsViewModel @Inject constructor(
 
     fun onUpdateThirdPartyDeposits() {
         viewModelScope.launch {
-            val currentThirdPartyDeposits = state.value.account?.onLedgerSettings?.thirdPartyDeposits
+            val currentThirdPartyDeposits = state.value.account?.onLedgerSettings?.thirdPartyDeposits ?: return@launch
 
-            val currentAssetExceptions = currentThirdPartyDeposits?.assetsExceptionList.orEmpty()
-            val currentDepositors = currentThirdPartyDeposits?.depositorsAllowList.orEmpty()
+            val newDepositRule = state.value.updatedThirdPartyDepositSettings?.depositRule ?: currentThirdPartyDeposits.depositRule
             val newAssetExceptions = state.value.updatedThirdPartyDepositSettings?.assetsExceptionList.orEmpty()
             val newDepositors = state.value.updatedThirdPartyDepositSettings?.depositorsAllowList.orEmpty()
 
-            val defaultRule = if (currentThirdPartyDeposits?.depositRule != state.value.updatedThirdPartyDepositSettings?.depositRule) {
-                checkNotNull(state.value.updatedThirdPartyDepositSettings?.depositRule)
-            } else {
-                null
-            }
-
-            ManifestPoet.buildThirdPartyDeposits(
-                settings = ManifestPoet.ThirdPartyDepositSettings(
-                    accountAddress = args.address,
-                    defaultDepositRule = defaultRule,
-                    removeAssetExceptions = currentAssetExceptions.minus(newAssetExceptions.toSet()),
-                    addAssetExceptions = newAssetExceptions.minus(currentAssetExceptions.toSet()),
-                    removeDepositors = currentDepositors.minus(newDepositors.toSet()),
-                    addedDepositors = newDepositors.minus(currentDepositors.toSet())
+            runCatching {
+                TransactionManifest.thirdPartyDepositUpdate(
```

Comment:
now sargon will compute diffs, which asset exceptions/depositors need to be removed and which will be added?

Reply:
Yes exactly